File: 기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int N = progresses.length;
        int[] answer = new int[N];
        
        int p = 0;
        int size = 0;
        while (p < N){
            int job = 100 - progresses[p];
            int time = job / speeds[p] + (job%speeds[p] > 0? 1: 0);
            
            int cnt = 0;
            boolean flag = true;
            for(int i=p; i<N; i++){
                progresses[i] += speeds[i] * time;
                
                if(flag && progresses[i] >= 100){
                    progresses[i] = 100;
                    
                    cnt++;
                    p++;
                }else
                    flag = false;
            }
            
            answer[size++] = cnt;
        }
        
        
        return Arrays.stream(answer).filter(n -> n!=0).toArray();
    }
}

File: 기능개발.py
def solution(progresses, speeds):
    answer = []
    
    N = len(progresses)
    p = 0
    
    while p < N:
        # 작업이 얼마나 남았는지, 몇일 뒤에 완료되는지 계산
        job = 100-progresses[p]
        time = job//speeds[p] if job % speeds[p] == 0 else job//speeds[p] + 1 
        
        cnt = 0
        flag = True
        for i in range(p, N):
            progresses[i] += speeds[i] * time
            
            if flag and progresses[i] >= 100:
                progresses[i] = 100
                p += 1
                cnt += 1
            else: 
                flag = False
        
        answer.append(cnt)
    
    return answer
